metadata/metadata_test.go

```diff
@@ -0,0 +1,326 @@
+			})
+		})
+		When("called with a pipelineRun containing nil Annotations", func() {
+			release := &v1alpha1.Release{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{},
+					Labels:      map[string]string{},
+				},
+			}
+			pipelineRun := &v1beta1.PipelineRun{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: nil,
+					Labels:      nil,
+				},
+			}
+			AddAnnotations(pipelineRun, release.GetAnnotations())
+			It("should create the Annotations map", func() {
+				Expect(pipelineRun.GetAnnotations()).ToNot(BeNil())
+			})
+		})
+	})
+
+	Context("metadata.AddLabels method", func() {
+		When("calls the addEntries function with an empty map", func() {
+			pipelineRun := &v1beta1.PipelineRun{
+				ObjectMeta: v1.ObjectMeta{
+					Labels: map[string]string{"pet/dog": "ruff"},
+				},
+			}
+			AddLabels(pipelineRun, map[string]string{})
+			It("src and dst maps should have the same length", func() {
+				Expect(len(pipelineRun.GetLabels())).To(Equal(1))
+			})
+		})
+		When("calls the addEntries function with a non-empty map", func() {
+			release := &v1alpha1.Release{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{},
+					Labels: map[string]string{
+						"pet/dog":   "bark",
+						"pet/cat":   "meow",
+						"pond/frog": "ribit",
+					},
+				},
+			}
+			pipelineRun := &v1beta1.PipelineRun{
+				ObjectMeta: v1.ObjectMeta{
+					Labels: map[string]string{"pet/dog": "ruff"},
+				},
+			}
+			AddLabels(pipelineRun, release.GetLabels())
+			It("should copy all src key/value pairs into dst that do not already exist", func() {
+				Expect(pipelineRun.GetLabels()["pet/dog"]).To(Equal("ruff"))
+				Expect(pipelineRun.GetLabels()["pet/cat"]).To(Equal("meow"))
+				Expect(pipelineRun.GetLabels()["pond/frog"]).To(Equal("ribit"))
+			})
+			It("should have increased the length of the dst map", func() {
+				Expect(len(pipelineRun.GetLabels())).To(Equal(3))
+			})
+		})
+		When("called with a pipelineRun containing nil Labels", func() {
+			release := &v1alpha1.Release{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{},
+					Labels:      map[string]string{},
+				},
+			}
+			pipelineRun := &v1beta1.PipelineRun{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: nil,
+					Labels:      nil,
+				},
+			}
+			AddLabels(pipelineRun, release.GetAnnotations())
+			It("should create the Annotations map", func() {
+				Expect(pipelineRun.GetLabels()).ToNot(BeNil())
+			})
+		})
+	})
+
+	//
+	// Get*WithPrefix methods
+	//
+	Context("metadata.GetAnnotationsWithPrefix method", func() {
+		When("calls the filterByPrefix function with an empty prefix string", func() {
+			release := &v1alpha1.Release{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{
+						"pet/dog":   "bark",
+						"pet/cat":   "meow",
+						"pond/frog": "ribit",
+					},
+					Labels: map[string]string{},
+				},
+			}
+			GetAnnotationsWithPrefix(release, "")
```

Comment:
shouldn't you be setting the return value of this equal to something and checking the length of that?